benchmark/flink-kafka/README.md

```diff
@@ -0,0 +1,129 @@
+# Running Nexmark Benchmarks on Flink with Kafka
+
+It's useful for comparative purposes to be able to run the same
+benchmark on DBSP and other systems.  These instructions explain how
+to run the [Nexmark benchmarks](https://github.com/nexmark/nexmark) on
+Flink in a reproducible manner.  The benchmarks run on a single
+physical node using multiple containers using Docker Compose.  One
+container runs the Flink jobmanager, and additional containers (by
+default 8) run Flink taskmanager.
+
+These instructions run Flink with Kafka as a data source.  **These
+instructions are unsatisfactory because they will use all your disk
+space quickly (1+ TB in 30 minutes).  Presumably that is fixable but
+it is not yet fixed.**
+
+The following instructions use `docker` and `docker-compose`, since
+that's what most people expect, but they were instead tested with
+`podman` and `podman-compose`, which are compatible.
+
+1. `cd` to the directory that contains this `README.md`, because the
+   files in this directory are useful for building the Docker
+   container and running Docker Compose.
+
+2. Download and unpack the Nexmark benchmark:
+
+```
+git clone https://github.com/nexmark/nexmark.git
+```
+
+3. Apply a couple of patches:
+
+```
+(cd nexmark && patch -p1 < ../avoid-jps-dependency.patch)
+(cd nexmark && patch -p1 < ../omit-q10.patch)
+```
+
+4. Build Nexmark:
+
+```
+(cd nexmark/nexmark-flink && ./build.sh)
+```
+
+   If it doesn't build, it might mean you've got the wrong version of
+   the JDK installed.  Nexmark and Flink only work with JDK 1.8 and
+   1.11, not newer versions.
+
+   This should output `nexmark/nexmark-flink/nexmark-flink.tgz`.  Be
+   warned that `build.sh` doesn't bail out on errors, so you might
+   still get something there even if the build partially fails.
+
+5. Extract Nexmark archive:
+
+```
+tar xzf nexmark/nexmark-flink/nexmark-flink.tgz
+```
+
+6. Download and unpack Kafka:
+
+```
+wget https://dlcdn.apache.org/kafka/3.4.0/kafka_2.13-3.4.0.tgz
+tar xfz kafka_2.13-3.4.0.tgz
+```
+
+7. Download the Flink connector for Kafka:
+
+```
+wget https://repo.maven.apache.org/maven2/org/apache/flink/flink-sql-connector-kafka/1.17.0/flink-sql-connector-kafka-1.17.0.jar
+```
+
+8. Build a Docker container based on the upstream Flink container that
+   also contains Nexmark and Kafka, and give it the tag
+   `nexmark-kafka`:
+
+```
+docker build -t nexmark-kafka .
+```
+
+7. Start the containers:
+
+```
+docker-compose -p nexmark up
+```
+
+   To reduce output spew, add `-d` (but sometimes that output can be
+   informative):
+
+```
+podman-compose up -d -p nexmark
+```
+
+8. Run the `insert_kafka` query to start inserting data into Kafka.
```

Comment:
We run all our Nexmark benchmarks with 100M input records.  Would the simplest solution to disk exhaustion be getting the generator to stop after 100M records (which should occupy ~10GB of disk space)

Reply:
That makes sense. The generator for Kafka just goes on forever (it's documented to work that way). It gets populated into kafka with a SQL query:

```
INSERT INTO kafka
SELECT event_type, person, auction, bid FROM datagen;
```

I wonder whether adding `LIMIT 100000000` is all it would take. I'll try it.

Reply:
> I wonder whether adding `LIMIT 100000000` is all it would take. I'll try it.

Even with a bunch more fussing, I can't seem to get this to work. Frustrating.

Reply:
Thanks, @blp , please let me know when it's ready to merge.